Userland/Libraries/LibWebView/SingletonProcessState.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Function.h>
+#include <AK/OwnPtr.h>
+#include <AK/Types.h>
+#include <LibCore/Socket.h>
+#include <LibIPC/ConnectionFromClient.h>
+#include <LibIPC/Forward.h>
+#include <LibIPC/MultiServer.h>
+#include <LibWebView/UIProcessClientEndpoint.h>
+#include <LibWebView/UIProcessServerEndpoint.h>
+
+namespace WebView {
+
+class UIProcessConnectionFromClient final
+    : public IPC::ConnectionFromClient<UIProcessClientEndpoint, UIProcessServerEndpoint> {
+    C_OBJECT(UIProcessConnectionFromClient);
+
+public:
+    virtual ~UIProcessConnectionFromClient() override = default;
+
+    virtual void die() override;
+
+    Function<void(Vector<ByteString> const& urls)> on_new_tab;
+    Function<void(Vector<ByteString> const& urls)> on_new_window;
+
+private:
+    UIProcessConnectionFromClient(NonnullOwnPtr<Core::LocalSocket>, int client_id);
+
+    virtual void create_new_tab(Vector<ByteString> const& urls) override;
+    virtual void create_new_window(Vector<ByteString> const& urls) override;
+};
+
+class SingletonProcessState {
```

Comment:
Maybe instead of `SingletonProcessState`, just call it `ChromeProcess`? It's all quite specific to the main Ladybird process, so a non-generic name sounds fine IMO